qctrlopencontrols/driven_controls/driven_control.py

```diff
@@ -545,42 +497,62 @@ def get_plot_formatted_arrays(self, coordinates=CARTESIAN, dimensionless_rabi_ra
         ArgumentsValueError
             Raised when an argument is invalid.
         """
+        if coordinates not in [CARTESIAN, CYLINDRICAL]:
+            raise ArgumentsValueError(
+                'Unsupported coordinates provided: ',
+                arguments={'coordinates': coordinates})
+
         if dimensionless_rabi_rate:
             normalizer = self.maximum_rabi_rate
         else:
             normalizer = 1
 
         if coordinates == CARTESIAN:
-            (x_amplitudes, y_amplitudes, detunings, times) = get_plot_data_from_segments(
-                np.vstack((self.amplitude_x / normalizer, self.amplitude_y / normalizer,
-                           self.detunings, self.durations)).T
-            )
-            plot_dictionary = {
-                'amplitudes_x': x_amplitudes,
-                'amplitudes_y': y_amplitudes,
-                'detunings': detunings,
-                'times': times
-            }
-
+            control_segments = np.vstack((
+                self.amplitude_x / normalizer,
+                self.amplitude_y / normalizer,
+                self.detunings,
+                self.durations)).T
         elif coordinates == CYLINDRICAL:
-            (x_plot, y_plot, detunings, times) = get_plot_data_from_segments(
-                np.vstack((self.rabi_rates / normalizer, self.azimuthal_angles,
-                           self.detunings, self.durations)).T
-            )
+            control_segments = np.vstack((
+                self.rabi_rates / normalizer,
+                self.azimuthal_angles,
+                self.detunings,
+                self.durations)).T
+
+        segment_times = np.insert(np.cumsum(control_segments[:, 3]), 0, 0.)
+        plot_time = (segment_times[:, np.newaxis] * np.ones((1, 2))).flatten()
```

Comment:
optional: since this trick for duplicating the arrays is used in a few places, and isn't super intuitive (to me, anyway) you could pull it out into a little private helper function (which could even be defined inside this method). That lets you save some code, and implicitly gives a little bit of documentation (in the function name).